### ISSUE TITLE
Bug: [no-base-to-string] hit URL object but should not

### ISSUE THREAD

**Issue by HolgerJeromin** [typescript-eslint/typescript-eslint#4999]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.1-rc&sourceType=module&code=BQOwpg7gBAogTnA9nYByARkg1mEqCU+AdAC6IDKJcAliAObD5A&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9AQPZYBGAhgnFhBQhOgQOYgBcIcKKZKEAF8gA&tsConfig=N4XyA

### Repro Code

```typescript
(new Error('broken')).toString()
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-base-to-string": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

The code should be fine as the runtime output is:
`'Error: broken'`

### Actual Result

eslint reports:
`'new Error('broken') will evaluate to '[object Object]' when stringified.ESLint (@typescript-eslint/no-base-to-string)`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.24.0` |
| `@typescript-eslint/parser`            | `5.24.0` |
  | `TypeScript`                           | `4.6.2` |
| `ESLint`                               | `8.15.0` |

**Comment by armano2** [comment]
issue has been solved upstream in typescript and its no longer incorrectly reported for typescript 4.6.4 and 4.7

**Comment by JoshuaKGoldberg** [comment]
Wowow great! 

I just checked and other built-ins such as `RegExp` still have the issue. But glad to see `Error` is fixed!

**Comment by armano2** [comment]
@JoshuaKGoldberg can you provide an example of error for ts 4.6+? i'm unable to reproduce it for URL, RegExp and Error

https://typescript-eslint.io/play/#ts=4.6.4&sourceType=module&code=BQOwpg7gBAqgSgGWAcgEYCcD2BrMJkCUBAdAC6YDKp6AliAObAEDcAsAFCiRRxj0CiADwAOKDDjyES5KrQZM2ncNH7os6MVlz4iZStTqMWHIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oCMBDZRLXxdk+KkwDm6KImjQO0SODABfECqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYwAviNUA

**Comment by bradzacher** [comment]
@armano2 - I think sometihng is wrong with lib loading?
[adding a `reference` comment reveals the error](https://typescript-eslint.io/play/#ts=4.6.4&sourceType=module&code=PQgEB4CcFMDNpgOwMbVAGwJYCMC8AiaAZwCYAGEgRn1GAD4BYAKGYApFoB3UAVQCUAMqwDk2SAHsA1tETCAlHIB0AF3EBlZZEyIA5qzkBuNh259oOgKIAPAA4ixUmfKWqNW3fqNN2XUBcgSkPYS0rIKKuqa2nqGzEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oCMBDZRLXxdk+KkwDm6KImjQO0SODABfECqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYwAviNUA)

[URL is not broken from the looks of it - it defines its own `toString`](https://github.com/microsoft/TypeScript/blob/b2617d6e38340b172058223879ec91964f9e5749/src/lib/dom.generated.d.ts#L14326)

[RegExp is broken](https://github.com/microsoft/TypeScript/blob/b2617d6e38340b172058223879ec91964f9e5749/src/lib/es5.d.ts#L925-L955), but we have it in the default ignore list.

[Error is similarly broken](https://github.com/microsoft/TypeScript/blob/b2617d6e38340b172058223879ec91964f9e5749/src/lib/es5.d.ts#L1007-L1011), and it is not ignored

**Comment by armano2** [comment]
thats interesting, I'm will have a look at this
